### PR TITLE
Haar 1202 email domains tests

### PR DESCRIPTION
**Pull request by olatomoloju**
Email domain listing integration tests.
Any takers?

### CODE REVIEW

File: integration-tests/e2e/emaildomain.manage.cy.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+context('EmailDomains', () => {
+  before(() => {
+    cy.clearCookies()
+  })
+
+  beforeEach(() => {
+    cy.task('reset')
+  })
+
+  it('Should display email domain listing with the create and delete email domain button and link respectively', () => {
```
Comment (karenmillermoj) [inline integration-tests/e2e/emaildomain.manage.cy.js:13]: We should also add in another test when there is an error

File: integration-tests/e2e/emaildomain.manage.cy.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const MenuPage = require('../pages/menuPage')
```
Comment (karenmillermoj) [inline integration-tests/e2e/emaildomain.manage.cy.js:1]: Are the other tests for email domain listing in a separate PR?

**Review by karenmillermoj — DISMISSED** [review]
lgtm

File: integration-tests/e2e/emaildomain.manage.cy.js (hunk 1 above)
Comment (olatomoloju) [reply integration-tests/e2e/emaildomain.manage.cy.js:13]: 1. Current test specifies min 10 and max 120 in auth > EmailDomainSpecification.kt. I can add that in.
2. Will look into CSRF.

File: integration-tests/e2e/emaildomain.manage.cy.js (hunk 2 above)
Comment (olatomoloju) [reply integration-tests/e2e/emaildomain.manage.cy.js:1]: Which other tests? There are separate tickets for create and delete func .

Comment (karenmillermoj) [reply integration-tests/e2e/emaildomain.manage.cy.js:1]: Tests for the getAllEmailDomains.js and any others that refer to the email domain listing

File: integration-tests/e2e/emaildomain.manage.cy.js (hunk 1 above)
Comment (olatomoloju) [reply integration-tests/e2e/emaildomain.manage.cy.js:13]: CSRF is applicable to the create domain screen and will be implemented in the create role ticket.
The "no results returned" test as been added.
Though the current auth test code checks for an email domain result count of 20-120, it appears irrelevant based on backend logic.

File: integration-tests/e2e/emaildomain.manage.cy.js (hunk 2 above)
Comment (olatomoloju) [reply integration-tests/e2e/emaildomain.manage.cy.js:1]: Additional test added.

File: backend/controllers/getAllEmailDomains.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        flash: jest.fn().mockReturnValue({ error: 'No email domains returned' }),
+      }
+      mockNoResultEmailDomainListingCall()
+      const render = jest.fn()
+      await getEmailDomainListing.index(reqWithError, { render })
+      expect(render).toBeCalledWith('emailDomainListing.njk', {
+        domains: noEmailDomains,
+        errors: { error: 'No email domains returned' },
+      })
+    })
```
Comment (karenmillermoj) [inline backend/controllers/getAllEmailDomains.test.js:80]: This should check [data-qa-errors] to show it is actually displayed

File: integration-tests/e2e/emaildomain.manage.cy.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      .emailDomainListingTableRows()
+      .eq(1)
+      .find('td')
+      .eq(2)
+      .should('include.html', '/delete-email-domain?id=acf5e424-2f7c-4bea-ac1e-07d2553f3e63&amp;name=DOMAIN2')
+    emailDomainListingPage.emailDomainListingTableRows().eq(1).should('include.text', 'Delete')
+
+    emailDomainListingPage.createEmailDomainButton().should('include.html', 'Add Email Domain')
+  })
+})
```
Comment (karenmillermoj) [inline integration-tests/e2e/emaildomain.manage.cy.js:42]: Should have a test ti check the error is displayed if no results

Comment (olatomoloju) [reply integration-tests/e2e/emaildomain.manage.cy.js:42]: That is covered in backend/controllers/getAllEmailDomains.test.js

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**